aws/data_source_aws_ecs_service.go

```diff
@@ -0,0 +1,83 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsEcsService() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEcsServiceRead,
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cluster_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"desired_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"launch_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"task_definition": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEcsServiceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecsconn
+
+	params := &ecs.DescribeServicesInput{
+		Cluster:  aws.String(d.Get("cluster_arn").(string)),
+		Services: []*string{aws.String(d.Get("service_name").(string))},
+	}
+
+	log.Printf("[DEBUG] Reading ECS Service: %s", params)
+	desc, err := conn.DescribeServices(params)
+
+	if err != nil {
+		return err
+	}
+
+	for _, service := range desc.Services {
```

Comment:
We generally prefer to check and immediately return errors for zero returned items or more than one returned item instead of using `for` loops in singular data sources. e.g.

```go
if len(desc.Services) == 0 {
  return fmt.Errorf("service with name %q in cluster %q not found", serviceName, clusterArn)
}

if len(desc.Services) > 1 {
  return fmt.Errorf("multiple services with name %q found in cluster %q", serviceName, clusterArn)
}

service := desc.Service[0]
d.SetId(aws.StringValue(service.ServiceArn))
// ...
```